onnx/checker.h

```diff
@@ -50,7 +50,7 @@ class CheckerContext final {
   bool is_main_graph() const {
     return is_main_graph_;
   }
-  void set_is_main_graph(bool is_main_graph) {
+  void set_is_main_graph(bool is_main_graph) const {
```

Comment:
this looks to me very bad. a "const" setter....

Reply:
Yes, I agree. I think we should be creating a copy of the CheckerContext to modify it. Furthermore, we can make all components of CheckerContext that do not change "const *" or "const &" so it can be shared (if desired).

Reply:
Or, it should be a non-const parameter.

Reply:
Agree.. Let's keep the ctx immutable, creating copy for subgraph.